core/src/main/java/org/infinispan/util/concurrent/CommandAckCollector.java

```diff
@@ -77,7 +77,7 @@ public void inject(
     * @param owners     the owners of the key. It assumes the first element as primary owner.
     * @param topologyId the current topology id.
     */
-   public void create(long id, Collection<Address> owners, int topologyId) {
+   public void create(long id, List<Address> owners, int topologyId) {
```

Comment:
If it's a set, you don't have any ordering guarantees so you can't assume the first element is the primary.